File: readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

// functions for each type
float letters (string input);
float words (string input);
float sentences (string input);

int main(void){

//get input
string input = get_string("Text: ");

//calculate each type
float letter = letters(input);
float word = words(input);
float sentence = sentences(input);

//printf("Letters: %f\nWords: %f\nSentences: %f\n", letter, word, sentence);

//calculation for coleman liau
float l = (letter / word) * 100;
float s = (sentence / word) * 100;
int index = 0.0588 * l - 0.296 * s - 15.8;

printf("Grade %i\n", index);

}

float letters (string input){
    int letter = 0;
    for (int i = 0, n=strlen(input); i < n; i++){
        if ((input[i] >= 'A' && input[i] <= 'Z') || (input[i] >= 'a' && input[i] <= 'z' )){
             letter++;}
    }
    return letter;
}

float words (string input){
    int word = 0;
    for (int i = 0, n=strlen(input); i < n; i++){
         if (input[i] == 32){
            word++;}
    }
    return word+=1;
}

float sentences (string input){
    int sentence = 0;
    for (int i = 0, n=strlen(input); i < n; i++){
         if ( input[i] == 46 || input[i] == 33 || input[i] == 63 ){
            sentence ++;}
    }
    return sentence;
    }

File: readability.py
import cs50

#calculate letters
def letters (inpt):
    letter = 0;
    for i in range (0, len(inpt), 1):
        if inpt[i] >= 'A' and inpt[i] <= 'Z' or inpt[i] >= 'a' and inpt[i] <= 'z':
             letter+=1
    
    return letter

#calculate words
def words (inpt):
    word = 0
    for i in range (0, len(inpt), 1):
         if ord(inpt[i]) == 32:
            word += 1
    
    return word


#calcultae sentences
def sentences (inpt):
    sentence = 0
    for i in range (0, len(inpt), 1):
         if ord(inpt[i]) == 46 or ord(inpt[i]) == 33 or ord(inpt[i]) == 63:
            sentence += 1
    
    return sentence
    

def main():
    
    inpt = cs50.get_string("Text: ");

    #calculate each type
    letter = letters(inpt);
    wordz = words(inpt);
    word = wordz + 1;
    sentence = sentences(inpt);

    print(f"Letters: {letter} Words: {word} Sentences: {sentence}");

    #calculation for coleman liau
    l = (letter / word) * 100;
    s = (sentence / word) * 100;
    indx = 0.0588 * l - 0.296 * s - 15.8;
    #find round
    index = round(indx);

    #print grade level
    
    if index < 1:
        print("Before Grade 1\n")
    elif index >= 1 and index <16:
        print(f"Grade {index}")
    elif index >= 16:
        print(f"Grade 16+")

if __name__ == "__main__":
    main()